### ISSUE TITLE
Broken Preview Mode? 

### ISSUE THREAD

**Issue by haakonfl** [bwydoogh/obsidian-force-view-mode-of-note#29]
Thanks for this plugin. I am unable though to set it up so that my dashboard note is in forced reader view while other notes I open are in edit view with live preview. That is my default setting. (My other notes do not have front matter data for this plugin.)

If I check «Ignore force view», other notes opens in reader view. If I uncheck that setting, other notes open in edit view with live preview disabled. Im I doing something wrong?

**Comment by bwydoogh** [comment]
Can you share your front matter?

```
---
obsidianUIMode: preview
---
```

...is all you need; it works fine here (navigate away from such note shows the note in "live preview" (edit) mode).

**Comment by pflenker** [comment]
I have the same issue.
Starting from a note with obsidianUIMode: preview, opening another note from this note opens in Reading Mode when Ignore Force View  is off, and in Source mode when it is on. It should be in live preview mode though when Ignore Force View is on.

**Comment by bwydoogh** [comment]
> It should be in live preview mode though when Ignore Force View is on.

Is the “Ignore opened files” toggle enabled? If so, opening another note from that given note with “obsidianUIMode: preview” will open in “preview” mode as expected. If the toggle is not enabled, the note will open in Obsidian’s default UI mode, which is likely “Live Preview”.

**Comment by pflenker** [comment]
Let me structure this a bit better. During reproduction, I noticed that it doesn't even matter where you start out from, even opening notes directly with no obsidianUIMode defined in the frontmatter brings up the editing mode in Source Mode instead of Live Preview Mode.


Obsidian Version: 1.2.8
Plugin Version: 1.1.1

Relevant Obsidian settings:
Default view for new tabs: Editing View
Default Editing mode: Live Preview

Plugin Settings: 
Ignore force view when not in front matter: false

How to reproduce:
Open any note that has obsidianUIMode not defined
EXPECTED BEHAVIOUR: The note is opened in editing mode / live preview mode
ACTUAL BEHAVIOUR: The note is opened in editing mode / source mode

**Comment by pflenker** [comment]
It seems like this line always falls through to "source": https://github.com/bwydoogh/obsidian-force-view-mode-of-note/blob/76e543452b7f8e429a3f27ba2b863bfc9aa895a3/main.ts#L103
I am not familiar with Plugin Development though, so I can't say for sure, but maybe this is a helpful pointer

**Comment by bwydoogh** [comment]
> Relevant Obsidian settings: Default view for new tabs: Editing View Default Editing mode: Live Preview
> 
> Plugin Settings: Ignore force view when not in front matter: false
> 
> How to reproduce: Open any note that has obsidianUIMode not defined EXPECTED BEHAVIOUR: The note is opened in editing mode / live preview mode ACTUAL BEHAVIOUR: The note is opened in editing mode / source mode

I'm using the same configuration, and I don't have this issue at all. Most of my notes don't have front matter, and they open as they should: live preview mode. 

The fallback to `source` only happens when `this.app.vault.config.defaultViewMode` has no value (which should never be the case). 

The only thing I can think of is perhaps a conflict with other plugins.

**Comment by pflenker** [comment]
I went ahead and added the following logging to the `main.js`, right after the declaration of `defaultViewMode`:
```
           console.log("defaultViewMode", defaultViewMode);
           console.log("this.app.vault.config.defaultViewMode", this.app.vault.config.defaultViewMode);
           console.log("this.app.vault.config", this.app.vault.config);
```

The output is:
```
defaultViewMode source
this.app.vault.config.defaultViewMode undefined
this.app.vault.config  {accentColor: ''}
```

**Comment by pflenker** [comment]
I have added my test vault to version control: https://github.com/pflenker/testvault/

**Comment by bwydoogh** [comment]
It is strange that `this.app.vault.config` only has `accentColor` as a property; I have a lot more properties (baseFontSize, alwaysUpdateLinks, cssTheme, foldHeading, and so on...; defaultViewMode is also one of the keys). That's the issue we have here.

I'll need to examine what is going on with `this.app.vault.config`.

**Comment by bwydoogh** [comment]
Released version `1.1.2` and this should resolve the issue. Please let me know if there are still any problems.

Thanks @mrenow
